File: ide/src/main/java/org/dau/ide/main/MainPane.java
package org.dau.ide.main;

import javafx.scene.Scene;
import javafx.scene.layout.BorderPane;
import javafx.scene.paint.Color;
import javafx.stage.Stage;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public final class MainPane extends BorderPane {

  public MainPane(MainTabPane tabPane, MainMenuBar menuBar) {
    super(tabPane, menuBar, null, null, null);
  }

  @Autowired
  public void initPrimaryStage(@MainQualifier Stage primaryStage) {
    var scene = new Scene(this, 800d, 600d);
    scene.setFill(Color.BLACK);
    primaryStage.setScene(scene);
  }
}

File: ide/src/main/kotlin/org/dau/ide/main/MainPane.kt
package org.dau.ide.main

import javafx.scene.Scene
import javafx.scene.layout.BorderPane
import javafx.scene.paint.Color
import javafx.stage.Stage
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class MainPane(tabPane: MainProjectTabs, menuBar: MainMenuBar) : BorderPane(tabPane, menuBar, null, null, null) {

  @Autowired
  fun initPrimaryStage(@MainQualifier primaryStage: Stage) {
    val scene = Scene(this, 800.0, 600.0)
    scene.fill = Color.BLACK
    primaryStage.scene = scene
  }
}
